AirCasting/SessionViews/ShareSession/ShareSessionViewModel.swift

```diff
@@ -0,0 +1,113 @@
+// Created by Lunar on 16/12/2021.
+//
+
+import Foundation
+
+protocol ShareSessionViewModel: ObservableObject {
+    var streamOptions: [ShareSessionStreamOptionViewModel] {get set}
+    var alert: AlertInfo? { get set }
+    var showSheet: Bool { get set }
+    var sharingLink: URL? { get set }
+    func didSelect(option: ShareSessionStreamOptionViewModel)
+    func shareLinkButtonGotPressed()
+}
+
+enum ShareSessionError: Error {
+    case noSessionURL
+}
```

Comment:
enum I think should be above protocol but not sure really @pawelgil could you please help here

Reply:
Also matter of taste. I usually put all types used by the protocol above it, yeah. As you cannot scope errors and structs inside protocol definition I use this kind-of rule of thumb: "If an enum/struct is something you'd put inside a protocol if it was possible, put it above it". IDK if this is clear - the reason is that you want a nice top to bottom readability in a file, so when you reach to the protocol you already know what errors mean and what return structs mean. This is also how it's done in majority of the projects afaik